### PR TITLE
Automatically add related tables when adding layers from GDAL datasets with embedded relationship information

### PR DESCRIPTION
**Pull request by nyalldawson**
For GDAL data formats with embedded relationship information, whenever a table is added we now discover these relationships and immediately add all the related tables to the current project (if they don't already exist), and auto-setup the corresponding QGIS relationships.

This works for all formats which support the GDAL relationship API, which include Geopackage (via the related tables extension),
ESRI FileGDB/OpenFileGDB, ESRI Personal Geodatabase, and sqlite/spatialite formats.

Requires GDAL 3.6

Sponsored by University of Uppsala

### CODE REVIEW

File: src/app/layers/qgsapplayerhandling.cpp [hunk 1]
```diff
@@ -136,8 +136,49 @@ void QgsAppLayerHandling::postProcessAddedLayer( QgsMapLayer *layer )
   }
 }
 
-void QgsAppLayerHandling::postProcessAddedLayers( const QList<QgsMapLayer *> & )
+void QgsAppLayerHandling::postProcessAddedLayers( const QList<QgsMapLayer *> &layers )
 {
+  for ( QgsMapLayer *layer : layers )
+  {
+    switch ( layer->type() )
```
Comment (lbartoletti) [inline src/app/layers/qgsapplayerhandling.cpp:143]: Is `layer` can be a nullptr?

**Review by lbartoletti — APPROVED** [review]

File: src/app/layers/qgsapplayerhandling.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/layers/qgsapplayerhandling.cpp:143]: No, those would be filtered out well before it hits this method

File: src/app/layers/qgsapplayerhandling.h [hunk 2]
```diff
@@ -157,6 +160,15 @@ class APP_EXPORT QgsAppLayerHandling
     //! Add a list of database layers to the map
     static QList< QgsMapLayer * > addDatabaseLayers( const QStringList &layerPathList, const QString &providerKey, bool &ok );
 
+    enum class DependencyFlag : int
```
Comment (elpaso) [inline src/app/layers/qgsapplayerhandling.h:163]: Can you add the documentation to this enum?

**Review by elpaso — APPROVED** [review]
Cool!

File: src/app/layers/qgsapplayerhandling.h (hunk 2 above)
Comment (nyalldawson) [reply src/app/layers/qgsapplayerhandling.h:163]: Done!

**Pull request merged**